src/compiler/checker.ts

```diff
@@ -13548,6 +13548,17 @@ namespace ts {
             return _jsxElementChildrenPropertyName;
         }
 
+        function createIntersectionOfApparentTypeOfJsxPropsType(propsType: Type): Type {
+            if (propsType && propsType.flags & TypeFlags.Intersection) {
+                const propsApprentType: Type[] = [];
+                for (const t of (<UnionOrIntersectionType>propsType).types) {
+                    propsApprentType.push(getApparentType(t));
+                }
+                return getIntersectionType(propsApprentType);
+            }
+            return propsType;

```

Comment:
what if it is not an intersection, you do not want to call getApparentType on it?